front/lib/production_checks/checks/scrub_deleted_core_document_versions.ts

```diff
@@ -33,66 +33,77 @@ export const scrubDeletedCoreDocumentVersionsCheck: CheckFunction = async (
 
   const storage = new Storage({ keyFilename: SERVICE_ACCOUNT });
 
-  const deletedDocumentsData = await coreReplica.query(
-    `SELECT * FROM data_sources_documents WHERE status = 'deleted'`
-  );
+  let pageId = 0;
+  let totalDeletedCount = 0;
 
-  const deletedDocuments = deletedDocumentsData[0] as {
-    id: number;
-    created: number;
-    data_source: number;
-    document_id: string;
-    hash: string;
-    timestamp: number;
-  }[];
+  do {
+    // paginate by id
+    const deletedDocumentsData = await coreReplica.query(
+      `SELECT * FROM data_sources_documents WHERE status = 'deleted' AND id > ${pageId} ORDER BY id LIMIT 1000`
```

Comment:
Actually yeah. Hadn't paid attention to the underlying logic that all docs retrieved were deleted, and basically applied classical id pagination 

but in this case you're right it's simpler to not have an offset, will update in a follow up PR

Reply:
Actually, I see in `scrubDocument` that the document is not always deleted from DB

This could be a logic flaw, which would explain the growing table, in which case we should fix by extracting the DELETE statement from `scrubDocument` and always performing it

Otherwise, if it's legit not to delete, we need to keep the pagination with id

Reply:
As discussed IRL, we keep as is since it's ~ legit not to delete the row if there's a most recent occurence
We can revisit the decision & dig if this turns out to be an issue to have a table consistently growing
Will rename to `pageId` to `lastSeenId` in another PR to make it clearer